### PR TITLE
[Quant][fx] Remove dequant-quant around getitem

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82675

Summary: https://github.com/pytorch/pytorch/issues/82480 saw
unnecessary dequant-quant pairs around the getitem op, which led
to significant slowdowns. This commit simply removes this pair in
the lowering step, since getitem already handles quantized inputs.

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_getitem_no_dequant_quant

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Tasks: https://github.com/pytorch/pytorch/issues/82480

Differential Revision: [D38427508](https://our.internmc.facebook.com/intern/diff/D38427508)

### CODE REVIEW

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 1]
```diff
@@ -931,7 +935,7 @@ def special_pattern_replacement(model: QuantizedGraphModule):
 
     return model
 
-def _lower_getattr_tensor_metadta_op(model: QuantizedGraphModule):
+def _lower_getattr_tensor_metadata_op(model: QuantizedGraphModule):
```
Comment (andrewor14) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:938]: (Unrelated change, just caught a small typo in neighboring lines)

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 2]
```diff
@@ -945,6 +949,24 @@ def _lower_getattr_tensor_metadta_op(model: QuantizedGraphModule):
             args[0] = n.args[0].args[0]
             n.args = tuple(args)
 
+def _lower_static_special_op(model: QuantizedGraphModule, qconfig_map: Dict[str, QConfigAny]):
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:952]: can we just add it to https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/_lower_to_native_backend.py#L94?

File: test/quantization/fx/test_quantize_fx.py [hunk 3]
```diff
@@ -6555,6 +6555,37 @@ def forward(self, x):
...
+                # x: [1, 5, 6]
+                x1 = x1 + inputs[:, :, -6:]
+                x2 = self.conv2(x1)
+                x2 = x2 + x1[:, :, -2:]
+                return x2
+
+        m = M()
+        m.eval()
+        qconfig_mapping = get_default_qconfig_mapping()
+        m = prepare_fx(m, qconfig_mapping, example_inputs=torch.rand(1, 5, 10))
```
Comment (jerryzh168) [inline test/quantization/fx/test_quantize_fx.py:6580]: I think we may want to check the input and output observer inserted for getitem are the same

**Review by jerryzh168 — APPROVED** [review]
looks good, can you add a check to make sure the input and output observer for getitem are the same instance?

File: test/quantization/fx/test_quantize_fx.py [hunk 4]
```diff
@@ -6555,6 +6559,55 @@ def forward(self, x):
         ])
         m3(*example_inputs)
 
+    def test_getitem_no_dequant_quant(self):
+        """
+        Test that we do not add dequant and quant ops around getitem.
```
Comment (jerryzh168) [inline test/quantization/fx/test_quantize_fx.py:6564]: nit: this needs to be updated

File: torch/ao/quantization/fx/prepare.py [hunk 5]
```diff
@@ -1312,7 +1312,24 @@ def insert_observers_for_model(
                         node_name_to_target_dtype, qconfig_map,
                         model, modules, graph)
 
-        #
+        # Second pass: Look for getitem nodes and make the input and output observers the same.
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/prepare.py:1315]: actually can we put this in a helper function?

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38427508).
